pkg/services/ngalert/store/alert_rule.go

```diff
@@ -228,33 +227,39 @@ func (st DBstore) UpdateAlertRules(ctx context.Context, rules []UpdateRule) erro
 }
 
 // GetOrgAlertRules is a handler for retrieving alert rules of specific organisation.
-func (st DBstore) GetOrgAlertRules(ctx context.Context, query *ngmodels.ListAlertRulesQuery) error {
+func (st DBstore) ListAlertRules(ctx context.Context, query *ngmodels.ListAlertRulesQuery) error {
 	return st.SQLStore.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
-		alertRules := make([]*ngmodels.AlertRule, 0)
-		q := "SELECT * FROM alert_rule WHERE org_id = ?"
-		params := []interface{}{query.OrgID}
+		q := sess.Table("alert_rule")
 
-		if len(query.NamespaceUIDs) > 0 {
-			placeholders := make([]string, 0, len(query.NamespaceUIDs))
-			for _, folderUID := range query.NamespaceUIDs {
-				params = append(params, folderUID)
-				placeholders = append(placeholders, "?")
-			}
-			q = fmt.Sprintf("%s AND namespace_uid IN (%s)", q, strings.Join(placeholders, ","))
+		if query.OrgID >= 0 {
```

Comment:
I still think that this should be a pointer too.  Is there a logic somewhere in the code that enforces this constraint (that orgID must be a non-negative)

Reply:
anyway, making the argument pointer emphasizes that it is optional, and I think it is important even in the internal code.

Reply:
I agree, as mentioned above I want our use of pointers and zero value to be consistent. I am thinking for this PR to be consistent with what we have know, and then we can change all optional conditions to pointers in a later PR?